fabric/src/main/java/com/discordsrv/fabric/console/executor/FabricCommandExecutor.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * This file is part of DiscordSRV, licensed under the GPLv3 License
+ * Copyright (c) 2016-2025 Austin "Scarsz" Shapiro, Henri "Vankka" Schubin and DiscordSRV contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.discordsrv.fabric.console.executor;
+
+import com.discordsrv.fabric.FabricDiscordSRV;
+import com.discordsrv.common.command.game.abstraction.executor.AdventureCommandExecutorProxy;
+import com.discordsrv.common.command.game.abstraction.executor.CommandExecutor;
+import net.kyori.adventure.text.Component;
+import net.minecraft.server.command.ServerCommandSource;
+
+import java.util.function.Consumer;
+
+public class FabricCommandExecutor implements CommandExecutor {
+
+    private final FabricDiscordSRV discordSRV;
+    private final ServerCommandSource source;
+
+    public FabricCommandExecutor(FabricDiscordSRV discordSRV, Consumer<Component> componentConsumer) {
+        this.discordSRV = discordSRV;
+        this.source = (ServerCommandSource) new AdventureCommandExecutorProxy(
+                discordSRV.getServer().getCommandSource(),
+                componentConsumer
+        ).getProxy();
```

Comment:
This won't work, the underlying API only lets interfaces be proxied and can't be cast like this.